lib/app/lifecycle.coffee

```diff
@@ -119,16 +119,21 @@ crypto = require 'crypto'
   next err
 
 @ssoAndRedirectBack = (req, res, next) ->
-  return res.send { success: true, user: req.user.toJSON() } if req.xhr
-  parsed = parse redirectBack req
-  parsed = parse resolve opts.APP_URL, parsed.path unless parsed.hostname
-  domain = parsed.hostname?.split('.').slice(1).join('.')
-  return redirectBack(req, res) if domain isnt 'artsy.net'
+  urlToRedirectTo = parse redirectBack req
+  urlToRedirectTo = parse resolve opts.APP_URL, urlToRedirectTo.path unless urlToRedirectTo.hostname
+  domain = urlToRedirectTo.hostname?.split('.').slice(1).join('.')
+  # return redirectBack(req, res) if domain isnt 'artsy.net'
```

Comment:
This line was introduced in https://github.com/artsy/artsy-passport/pull/80/commits/f0c51b8addc89d4ab2111c49517a77d119149344 (PR: https://github.com/artsy/artsy-passport/pull/80) by Craig.

Does anyone recall the rational for this short-circuit return? It makes it harder to develop against artsy-passport, so I would prefer to remove it if possible.

cc: @mzikherman @sweir27 (who created some PRs which referenced #80)

Reply:
Seems like it might be security-related but I don't fully understand. If you do drop this line, the line above can go away as well.

Reply:
I can't recall anything specific about this (@dleve123 do you have those PRs you mention handy?). It does appear to be security-related, but I suppose we could add some kind of environment-specific condition to avoid the developer experience issue.

That being said, I'm not sure when we'd encounter this situation?

Reply:
> (@dleve123 do you have those PRs you mention handy?)

https://github.com/artsy/prediction/pull/208, https://github.com/artsy/microgravity-private/pull/1401 (you didn't create them, just were assigned to them, my b)

> It does appear to be security-related, but I suppose we could add some kind of environment-specific condition to avoid the developer experience issue.

So both of this repo and force is open source  – perhaps there's a vector where someone could run a local force, and, given a valid set of auth credentials to gravity (which isn't open source), get auth'ing redirect link to gravity? Either way, it's really straightforward to spoof being on the artsy domain (there are in fact steps for this in this project's [README](https://github.com/artsy/artsy-passport/blob/master/README.md#contributing)).

Reply:
I believe the intent is to only sign someone in to gravity, if they are logging in to Force (or any other Artsy app, which uses artsy-passport).

Theoretically this lib is OSS so you might use this for your own client application (not an Artsy app, not on artsy.net). In that case, you probably wouldn't want this (since the only convenience is to be able to login to _other_ apps).

I don't know if it's security-related per se, and more just expected behavior.

** Caveat- this is just my speculation!

Reply:
Got it, thanks for the context all! Don't want to dedicate a ton of mind-share to this and seems like we don't have a great reason to keep this limitation in, so opting to remove this LOC. FWIW, `yarn test` passes without it.